### PR TITLE
feat: add a CLI message when newer version is available

### PR DESCRIPTION
**Pull request by yuliabaron**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Adding a message to CLI when a newer version is available

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/SC-6399

#### Screenshots
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -110,5 +111,6 @@
       "help/**/*.txt"
     ]
   },
-  "snyk": true
+  "snyk": true,
+  "version": "1.109.0"
```
Comment (darscan) [inline package.json]: Howdy! We should not commit the version number into our package.json, semantic release will take care of that during the publish step.

File: src/cli/index.ts [hunk 2]
```diff
@@ -14,6 +14,19 @@ import errors = require('../lib/error');
...
 import {isPathToPackageFile} from '../lib/detect';
 
+import * as updateNotifier from 'update-notifier';
+var pkg = require('../../package.json');
+
+async function updateCheck() {
+    // Checks for available update and returns an instance
+    // Default updateCheckInterval is once a day
+    var notifier = updateNotifier({
+        pkg: pkg,
```
Comment (miiila) [inline src/cli/index.ts]: You can write this as `updateNotifier({pkg}); when object key is same as a variable it points to, you can use this syntactic sugar :-)

File: src/cli/index.ts [hunk 3]
```diff
@@ -14,6 +14,19 @@ import errors = require('../lib/error');
...
+    });
+    // Notify using the built-in convenience method
```
Comment (miiila) [inline src/cli/index.ts]: I'm not sure I understand this comment. Do you think we need it?

File: src/cli/index.ts [hunk 4]
```diff
@@ -14,6 +14,19 @@ import errors = require('../lib/error');
 import ansiEscapes = require('ansi-escapes');
 import {isPathToPackageFile} from '../lib/detect';
 
+import * as updateNotifier from 'update-notifier';
+var pkg = require('../../package.json');
+
+async function updateCheck() {
```
Comment (miiila) [inline src/cli/index.ts]: As notify is not asycnhronous (i.e. we don't do `await notifier.notify()`, `async` keyword before the function name is misleading.

File: src/cli/index.ts [hunk 5]
```diff
@@ -108,6 +121,7 @@ function checkPaths(args) {
 }
 
 async function main() {
+  await updateCheck();
```
Comment (miiila) [inline src/cli/index.ts]: Related to the comment above, `updateCheck()` can be fully synchronous.

File: src/cli/index.ts [hunk 6]
```diff
@@ -14,6 +14,19 @@ import errors = require('../lib/error');
 import ansiEscapes = require('ansi-escapes');
 import {isPathToPackageFile} from '../lib/detect';
 
+import * as updateNotifier from 'update-notifier';
+var pkg = require('../../package.json');
```
Comment (miiila) [inline src/cli/index.ts]:
```diff
- var
+ const
```

File: src/cli/index.ts (hunk 2 above)
Comment (miiila) [inline src/cli/index.ts]:
```diff
- var
+ const
```

File: src/cli/index.ts (hunk 6 above)
Comment (miiila) [inline src/cli/index.ts]: Personal sidenote - I don't like using `require` to load .json files, I would suggest `fs.readFile`, but I fear we can saw this pattern in this app more often, so it's probably good to go.

File: src/cli/index.ts [hunk 7]
```diff
@@ -14,6 +14,19 @@ import errors = require('../lib/error');
...
+
+async function updateCheck() {
+    // Checks for available update and returns an instance
```
Comment (miiila) [inline src/cli/index.ts]: I'm not sure about this comment either - do you think it brings in something important?

File: src/cli/index.ts [hunk 8]
```diff
@@ -13,6 +13,34 @@ import spinner = require('../lib/spinner');
...
+    const pkgFile = JSON.parse(fs.readFileSync(pkgPath, 'utf8'));
+
+    // Checks for available update and returns an instance
+    // Default updateCheckInterval is once a day
+    // set the try/catch to catch the errors if for some reason package.json doesn't have the required info
+    let notifier;
+    try {
+      notifier = updateNotifier({pkgFile});
+    } catch (err) {
+      console.warn('Failed to start updater, ERR=', err);
```
Comment (yuliabaron) [inline src/cli/index.ts]: Not sure whether to keep this warning. The code works from different directories, errors were observed during the tests on travis only

Comment (darscan) [reply src/cli/index.ts]: I imagine that we get here when `version` is not present in the package.json as would be the case on Travis etc.

Comment (yuliabaron) [reply src/cli/index.ts]: exactly

File: src/cli/index.ts [hunk 9]
```diff
@@ -13,6 +13,34 @@ import spinner = require('../lib/spinner');
...
+      return;
+    }
+
+    notifier.notify();
```
[the remark this answers is not in the document]
Comment (yuliabaron) [reply src/cli/index.ts]: i return from catch, but you're right, will move

**Pull request merged**